### ISSUE TITLE
WinUI 3: AppCenter includes dependency to UniversalWindowsPlatform

### ISSUE THREAD

**Issue by Marv51** [microsoft/appcenter-sdk-dotnet#1759]
### **Description**
A packaged WinUI 3 app with AppCenter loads the dependencies for the wrong platform.

For WinUI 3 app with a packaging project, the AppCenter dependency is depending on the UAP/UWP version of AppCenter. 
This means that AppCenter will include `Microsoft.NETCore.UniversalWindowsPlatform` as a transitive dependency. This dependency is massive and is otherwise not needed. The nuget package folder with AppCenter for my app is a bit over 4GB without AppCenter it is only 1.5GB. 
I believe that these dependencies might not even be included in the build output, but the download of so much data really slows down my CI environment. (Even with caching)

### **Repro Steps**

1. Create a new blank WinUI 3 App for C# with the template: "Blank App, Packaged with WAP" 
2. To see transitive dependencies, add `<RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>` to the .wappproj and .csproj files.
3. Note or copy both the packages.lock.json files for the app and the packaging project. To compare before and after AppCenter is installed.
4. Add Microsoft.AppCenter.Crashes nuget package the usual way to the project.
5. build and run
6. note that the packages.lock.json file includes a lot more entries, including "Microsoft.NETCore.UniversalWindowsPlatform".

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    - Windows 11 with WinUI 3
2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
    - does not apply
2. Which SDK version are you using?
    - 5.0.3
8. Which OS version did you experience the issue on?
    - Windows 11 23H2
9. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Windows Laptop
10. What third party libraries are you using? 
    - none
11. Please enable verbose logging for your app using `AppCenter.LogLevel = LogLevel.Verbose` before your call to `AppCenter.Start(...)` and include the logs here:
    - not relevant imho, because it is a dependency issue, can deliver if needed

**Comment by Marv51** [comment]
I have removed the `UAP` platform support from the nuget packages locally and removed any dependencies for the UAP platform using Nuget Package Explorer. 
When I use these packages the problem is gone. A clean restore of the solution is way fast and requires 2GB less downloads.

**Comment by MikhailSuendukov** [comment]
Hi @Marv51 and thank you for reaching out to us. I would like to clarify whether you were able to completely solve your problem?

**Comment by Marv51** [comment]
When I produce my own nuget packages without any UWP/UAP content. The problem is resolved, but that is of course not really a solution that is very sustainable.

**Comment by MikhailSuendukov** [comment]
Sorry for the delay, but I was unable to reproduce this issue, could you please provide us with a small demo project on which this error will be reproduced?

**Comment by Marv51** [comment]
[AppCenterSDKRepro.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13067781/AppCenterSDKRepro.zip)

Go to your .nuget/packages folder. Delete all all content. The nuget folder is usually in your C:/Users/YourName profile folder (maybe going to %NUGET_PACKAGES% works too).

When you unzip this project, and open the solution in Visual Studio 2022, make sure the "AppSDKRepro (Package)" is selected as "Set as Startup Project". I select "x64" and click "Run". Wait for the sample window with a test button to appear. 

Observe that the project first restores about 1.5GB. But when you run the project, until a Window with a button appears the packages folder contains 4.1GB.

<img width="432" alt="after" src="https://github.com/microsoft/appcenter-sdk-dotnet/assets/471338/c9f38eb9-3dc8-4f3f-adcf-e7b3dfb673d9">

**Comment by MikhailSuendukov** [comment]
I apologize for the delay, I was able to reproduce this behavior, but when checking the size of the microsoft.netcore.universalwindowsplatform folder, it turned out that its size no more than 30 megabytes. Is this critical for you?

**Comment by Marv51** [comment]
The microsoft.netcore.universalwindowsplatform folder may be the only direct dependency and it is not giant. However, it pulls in many more dependencies. The size of these dependencies on disk is about 2.5GB. I find the `packages.lock.json` help me understand the dependencies a lot better.

And I think I mentioned it above, on a dev-machine that is not optimal, but acceptable. 
However, this significantly slows down our CI builds. Meaning money and flexibility is lost. It makes caching of nuget dependencies in GitHub-Actions almost impossible, because the cache is only 5GB. 

Look at this small extract from the top of tree that AppCenter pulls in:
```json
      "Microsoft.AppCenter": {
        "type": "Transitive",
        "resolved": "5.0.3",
        "contentHash": "[...]",
        "dependencies": {
          "Microsoft.NETCore.UniversalWindowsPlatform": "6.2.8",
          "Newtonsoft.Json": "13.0.2",
          "SQLitePCLRaw.bundle_green": "2.1.5"
        },
},
      "Microsoft.NETCore.UniversalWindowsPlatform": {
        "type": "Transitive",
        "resolved": "6.2.8",
        "contentHash": "[...]",
        "dependencies": {
          "Microsoft.NETCore.Platforms": "2.1.0",
          "Microsoft.Net.Native.Compiler": "2.2.3",
          "Microsoft.Net.UWPCoreRuntimeSdk": "2.2.8",
          "NETStandard.Library": "2.0.3"
        },
},
"Microsoft.Net.Native.Compiler": {
        "type": "Transitive",
        "resolved": "2.2.3",
        "contentHash": "[...]",
        "dependencies": {
          "runtime.win10-arm.Microsoft.Net.Native.Compiler": "2.2.3",
          "runtime.win10-arm64.Microsoft.Net.Native.Compiler": "2.2.3",
          "runtime.win10-x64.Microsoft.Net.Native.Compiler": "2.2.3",
          "runtime.win10-x86.Microsoft.Net.Native.Compiler": "2.2.3"
        }
      },
"Microsoft.Net.UWPCoreRuntimeSdk": {
        "type": "Transitive",
        "resolved": "2.2.8",
        "contentHash": "[...]",
        "dependencies": {
          "runtime.win10-arm.Microsoft.Net.UWPCoreRuntimeSdk": "2.2.8",
          "runtime.win10-x64.Microsoft.Net.UWPCoreRuntimeSdk": "2.2.8",
          "runtime.win10-x86.Microsoft.Net.UWPCoreRuntimeSdk": "2.2.8"
        }
      },

**Comment by MikhailSuendukov** [comment]
I apologize for the long delay. We looked at your problem and realized that apparently the problem lies in the WAP wrapper of the application. Because when running a WinUI application without a wrapper, the UWP transitive dependency does not appear in package.lock. Most likely, it is the WAP wrapper that pulls up these transitive dependencies, since it may require a UWP target for a reverse dependency or for implementing some functionality that relates to it. In our nuspec files, all transitive dependencies are divided into target frameworks, so the WinUI application should not use the UWP framework. If you have any ideas regarding this it would be very helpful.

**Comment by Marv51** [comment]
I tried looking deeper into this with binlog and structured log viewer but did not find any new clues why this is happening.

The way I see most nuget packages handle this is that they have separate nuget packages for UWP and WinUI. I understand that might not be an option for you. Personally I think making a separate WinUI 3 AppCenter package would very interesting.

I think I might build my own packages for now, then I can only include the relevant platforms.

I guess this is likely something the WinUI team needs to fix. You might have more influence on them as an internal/partner request.

**Comment by MikhailSuendukov** [comment]
Thank you for sharing this idea with us. We will monitor to see if others encounter similar problems. In the meantime, we'd appreciate it if you could keep us updated on your progress with building the packages you mentioned.

**Comment by Marv51** [comment]
So I realize that this is not really helpful, but I created a fork here: https://github.com/Marv51/appcenter-sdk-dotnet

I ripped out anything that is not needed for Crashes & Analytics for a modern packaged WinUI app.

As soon as the UWP target is removed from the nuget package this issue is resolved.

I would contribute some stuff from my fork to this repo, but I think the support for various dotnet versions makes this difficult.
